contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,910 @@
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        deadline: block.timestamp
+                    })
+                );
+
+            _wethAmountSupplied = __wethAmountSupplied;
+            _oethbAmountSupplied = __oethbAmountSupplied;
+        }
+
+        _updateUnderlyingAssets();
+        emit LiquidityAdded(
+            _wethBalance, // wethAmountDesired
+            _oethbRequired, // oethbAmountDesired
+            _wethAmountSupplied, // wethAmountSupplied
+            _oethbAmountSupplied, // oethbAmountSupplied
+            tokenId, // tokenId
+            underlyingAssets
+        );
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+        positionManager.approve(address(clGauge), tokenId);
+        clGauge.deposit(tokenId);
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /**
+     * @dev Check that the Aerodrome pool price is within the expected
+     *      parameters.
+     *      This function ignores whether the strategy contract has liquidity
+     *      position in the pool.
+     */
+    function _checkForExpectedPoolPrice() internal {
+        uint160 _currentPrice = getPoolX96Price();
+        // check we are in inspected tick range
+        require(
+            _currentPrice >= sqrtRatioX96Tick0 &&
+                _currentPrice <= sqrtRatioX96Tick1,
+            "Not in expected tick range"
+        );
+
+        // If token addresses were reversed estimateAmount0 would
+        // be required here
+        uint256 _normalizedWethAmount = 1 ether;
+        uint256 _correspondingOETHAmount = helper.estimateAmount1(
+            _normalizedWethAmount,
+            address(clPool),
+            _currentPrice,
+            lowerTick,
+            upperTick
+        );
+
+        // 18 decimal number expressed weth tick share
+        uint256 _wethSharePct = _normalizedWethAmount.divPrecisely(
+            _normalizedWethAmount + _correspondingOETHAmount
+        );
+
+        uint256 _wethDiff = Math.max(poolWethShare, _wethSharePct) -
+            Math.min(poolWethShare, _wethSharePct);
+
+        if (_wethDiff < poolWethShareVarianceAllowed) {
+            emit PoolRebalanced(_wethSharePct, poolWethShare);
+        } else {
+            revert PoolRebalanceOutOfBounds(_wethSharePct, poolWethShare);
+        }
+    }
+
+    /**
+     * Burns any OETHb tokens remaining on the strategy contract
+     */
+    function _burnOethbOnTheContract() internal {
+        uint256 _oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        if (_oethbBalance > 0) {
+            IVault(vaultAddress).burnForStrategy(_oethbBalance);
+        }
+    }
+
+    /// @dev this function assumes there are no uncollected tokens in the clPool owned by the.
+    ///      strategy contract. For that reason any liquidity withdrawals also collect the tokens.
+    function _updateUnderlyingAssets() internal {
+        if (tokenId == 0) {
```

Comment:
⚪️ I really prefer the fast return style for the exceptional case. So this could return after setting underlyingAssets in the zero token case.

Reply:
agreed thanks. Done here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/e931cb4b4089c52f98d45ba2f5008f07ac7bf220